### PR TITLE
curvefs/client: make user agent configurable

### PR DESCRIPTION
**Pull request by Ziy1-Tan**
### What problem does this PR solve?

Issue Number: https://github.com/opencurve/curve/issues/2499

Problem Summary: 

Let user agent item configurable for users having security reasons

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curve-ansible/roles/generate_config/templates/s3.conf.j2 [hunk 1]
```diff
@@ -10,7 +10,7 @@ s3.sk={{ s3_sk }}
 # http = 0, https = 1
 s3.http_scheme={{ s3_http_scheme }}
 s3.verify_SSL={{ s3_verify_ssl }}
-s3.user_agent_conf={{ s3_user_agent_conf }}
```
Comment (Cyber-SiKu) [inline curve-ansible/roles/generate_config/templates/s3.conf.j2:13]: Ansible is no longer maintained and can not be modified

File: curve-ansible/roles/generate_config/defaults/main.yml [hunk 2]
```diff
@@ -265,7 +265,7 @@ nebd_server_response_return_rpc_when_io_error: false
 # s3配置默认值
 s3_http_scheme: 0
 s3_verify_ssl: false
-s3_user_agent_conf: S3 Browser
```
Comment (Cyber-SiKu) [inline curve-ansible/roles/generate_config/defaults/main.yml:268]: ditto

**Comment by Ziy1-Tan** [comment]
cicheck

**Review by Cyber-SiKu — COMMENTED** [review]
Have you tested that it works?

File: curve-ansible/roles/generate_config/templates/s3.conf.j2 (hunk 1 above)
Comment (Ziy1-Tan) [reply curve-ansible/roles/generate_config/templates/s3.conf.j2:13]: Ok.

**Comment by Ziy1-Tan** [comment]
> Have you tested that it works?

Yes. I made sure that 
1. `s3.user_agent` appears in any other configuration item such as `s3.verify_SSL`. 
2. By debugging `TEST_F(S3CompactTest, test_S3AdapterManager)`, I observed that the `userAgent` item has been assigned correctly.

**Comment by Ziy1-Tan** [comment]
cicheck

**Review by ilixiaocui — APPROVED** [review]

**Comment by Ziy1-Tan** [comment]
cicheck

File: curve-ansible/roles/generate_config/defaults/main.yml (hunk 2 above)
Comment (Ziy1-Tan) [reply curve-ansible/roles/generate_config/defaults/main.yml]: Fixed.

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by wuhongsong** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
curvefs/conf/client.conf
This file should be the configuration file of the client of curvefs and should also need to be modified.

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
> curvefs/conf/client.conf This file should be the configuration file of the client of curvefs and should also need to be modified.

Get it. I'm trying to figure out why the mds election is timeout but I'm failed.

**Review by wuhongsong — DISMISSED** [review]

**Comment by wuhongsong** [comment]
> > curvefs/conf/client.conf This file should be the configuration file of the client of curvefs and should also need to be modified.
> 
> Get it. I'm trying to figure out why the mds election is timeout but I'm failed.

@SeanHai   help

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
@Ziy1-Tan 

2023-06-21 10:06:04.670070 I | [587688dba922b1f6/10.182.2.26:16701]stop watch current leader
F 2023-06-21T10:06:04.680009+0800     6 s3_adapter.cpp:85]
*** Check failure stack trace: ***
*** Aborted at 1687313164 (unix time) try "date -d @1687313164" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x6) received by PID 6 (TID 0x7fcbd210d4c0) from PID 6; stack trace: ***
    @     0x7fcbd014e0e0 (unknown)
    @     0x7fcbce805fff gsignal
    @     0x7fcbce80742a abort
    @     0x55bec06199ec google::FlushAndAbort()
    @     0x55bec0616c3e google::LogMessage::Fail()
    @     0x55bec0616b87 google::LogMessage::SendToLog()
    @     0x55bec06163b0 google::LogMessage::Flush()
    @     0x55bec0619380 google::LogMessageFatal::~LogMessageFatal()
    @     0x55bebfde1858 curve::common::InitS3AdaptorOptionExceptS3InfoOption()
    @     0x55bebf94fe43 curvefs::mds::MDS::InitFsManagerOptions()
    @     0x55bebf950179 curvefs::mds::MDS::Init()
    @     0x55bebf94c42a main
    @     0x7fcbce7f32e1 __libc_start_main
    @     0x55bebf94bc4a _start
    @                0x0 (unknown)
fileList: []
WARNING: Logging before InitGoogleLogging() is written to STDERR

The curvefs/conf/mds.conf also need add this conf item. If not, it will crash when init s3 adaptor in mds of curvefs.

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cichek

**Comment by caoxianfei1** [comment]
image

add more test case @Ziy1-Tan

**Comment by wuhongsong** [comment]
cicheck

**Comment by wuhongsong** [comment]
> image
> add more test case @Ziy1-Tan

I have add some test case，lets try it again.

**Review by Cyber-SiKu — APPROVED** [review]

**Comment by SeanHai** [comment]
cicheck

**Comment by wuhongsong** [comment]
@Ziy1-Tan  will do it later, It's a bit busy these days.

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

**Pull request merged**